collectors/python.d.plugin/riak/riak.chart.py

```diff
@@ -0,0 +1,528 @@
+        ]
+    },
+    "core.pbc_active": {
+        "options": [None, "Number of currently active protocol buffer connections", "connections", "load", "riak.core.protobuf_connections", "line"],
+        "lines": [
+            ["pbc_active", "active", "absolute"],
+            # ["pbc_connects", "established_pastmin", "absolute"]
+        ]
+    },
+    "core.read_repairs": {
+        "options": [None, "Number of read repair operations this node has coordinated", "repairs", "load", "riak.core.repairs", "line"],
+        "lines": [
+            ["read_repairs", "read", "absolute"]
+        ]
+    },
+    "core.fsm_active": {
+        "options": [None, "Active finite state machines by kind", "fsms", "load", "riak.core.fsm_active", "line"],
+        "lines": [
+            ["node_get_fsm_active", "get", "absolute"],
+            ["node_put_fsm_active", "put", "absolute"],
+            ["index_fsm_active", "secondary index", "absolute"],
+            ["list_fsm_active", "list keys", "absolute"]
+        ]
+    },
+    "core.fsm_rejected": {
+        # Writing "Sidejob's" here seems to cause some weird issues: it results in this chart being rendered in
+        # its own context and additionally, moves the entire Riak graph all the way up to the top of the Netdata
+        # dashboard for some reason.
+        "options": [None, "Finite state machines being rejected by Sidejobs overload protection", "fsms", "load", "riak.core.fsm_rejected", "line"],
+        "lines": [
+            ["node_get_fsm_rejected", "get", "absolute"],
+            ["node_put_fsm_rejected", "put", "absolute"]
+        ]
+    },
+
+    # General Riak Search Load / Health metrics
+    "search.errors": {
+        "options": [None, "Number of writes to Search failed due to bad data format by reason", "writes", "load", "riak.search.index", "line"],
+        "lines": [
+            ["search_index_bad_entry_count", "bad_entry", "absolute"],
+            ["search_index_extract_fail_count", "extract_fail", "absolute"],
+        ]
+    }
+}
+
+
+class Service(UrlService):
+    def __init__(self, configuration=None, name=None):
+        UrlService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+        self.keys = list(KEYS)
+        for k in ["mean", "median", "95", "99", "100"]:
+            for m in STAT_KEYS:
+                self.keys.append(m + "_" + k)
+        for k in ["min", "max", "mean", "median", "95", "99", "999"]:
+            for m in SEARCH_LATENCY_KEYS:
+                self.keys.append(m + "_" + k)
+        for k in ["min", "max", "mean", "median", "total"]:
+            for m in VNODEQ_KEYS:
+                self.keys.append(m + "_" + k)
+
+    def _get_data(self):
+        """
+        Format data received from http request
+        :return: dict
+        """
+        raw = self._get_raw_data()
+        if not raw:
+            return None
+
+        data = dict()
+        stats = loads(raw)
```

Comment:
use `try, except` for loads `(TypeError, ValueError)`

Reply:
Wrapped in `try` / `except` in fa4e72e18b237a1333f8c2aad2e66762eb47940d. Is there anything else I should do in case of error besides `return None`?

Reply:
no :+1: